crypto/bio/b_sock.c

```diff
@@ -364,6 +397,39 @@ static void ghbn_free(struct hostent *a)
 
 # endif
 
+int BIO_gethostbyname_r(const char *name, struct hostent *he,
+               struct hostent **result)
+{
+    char buf[GETHOSTNAME_R_BUF];
+    int h_errnop;
+    int err;
+    char *tmp_buf = NULL;
+
+    *result = NULL;
+    err = gethostbyname_r(name, he, buf, sizeof(buf), result,
+                 &h_errnop);
+    if (err == ERANGE) {
```

Comment:
I wouldn't worry about using malloc on failure.  Try with the one buffer size on the stack and fail if that's not sufficient.
For reference, glibc 2.24 uses 1024 bytes for the `gethostbyname(3)` temporary buffer.